packages/server/test/support/helpers/e2e.js

```diff
@@ -232,60 +257,74 @@ const getMochaItFn = function (only, skip, browser, specifiedBrowser) {
   return it
 }
 
-const getBrowsers = function (generateTestsForDefaultBrowsers, browser, defaultBrowsers) {
-  // if we're generating tests for default browsers
-  if (generateTestsForDefaultBrowsers) {
-    // then return an array of default browsers
-    return defaultBrowsers
+function getBrowsers (browserPattern) {
+  if (!browserPattern.length) {
+    return DEFAULT_BROWSERS
   }
 
-  // but if we haven't been told to generate tests for default browsers
-  // and weren't provided a specified browser then throw
-  if (!browser) {
-    throw new Error('A browser must be specified when { generateTestsForDefaultBrowsers: false }.')
+  let selected = []
+
+  const addBrowsers = _.clone(browserPattern)
+  const removeBrowsers = _.remove(addBrowsers, (b) => b.startsWith('!')).map((b) => b.slice(1))
+
+  if (removeBrowsers.length) {
+    selected = _.without(AVAILABLE_BROWSERS, ...removeBrowsers)
+  } else {
+    selected = _.intersection(AVAILABLE_BROWSERS, addBrowsers)
   }
 
-  // otherwise return the specified browser
-  return [browser]
+  if (!selected.length) {
+    throw new Error(`options.browser: "${browserPattern}" matched no browsers`)
+  }
+
+  return selected
 }
 
-const localItFn = function (title, options = {}) {
-  options = _
-  .chain(options)
-  .clone()
-  .defaults({
+const normalizeToArray = (value) => {
+  if (value && !_.isArray(value)) {
+    return [value]
+  }
+
+  return value
+}
+
+const localItFn = function (title, opts = {}) {
+  opts.browser = normalizeToArray(opts.browser)
+  opts.stdoutInclude = normalizeToArray(opts.stdoutInclude)
+  opts.stdoutExclude = normalizeToArray(opts.stdoutExclude)
```

Comment:
i don't really like having either of these two methods...

there is already a callback function: `options.onStdout` that yields you the `stdout` string so you can write normal assertions. 

i feel like its more explicit and easier to just do that than abstract this here. what if you wanted to use a `regexp` instead of a string?